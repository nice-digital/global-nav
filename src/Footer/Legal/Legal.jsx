import styles from "./Legal.module.scss";
import TrackedLink from "../../TrackedLink";
import { footerClickEventAction } from "../../tracker";

const Legal = () => {
	return (
		<div className={styles.wrapper}>
			<div className={styles.container}>
				<nav className={styles.menu} aria-label="Legal menu">
					<ul>
						<li>
							<TrackedLink
								href="https://www.nice.org.uk/accessibility"
								eventAction={footerClickEventAction}
								eventLabel="Accessibility"
							>
								Accessibility
							</TrackedLink>
						</li>
						<li>
							<TrackedLink
								href="https://www.nice.org.uk/freedom-of-information"
								eventAction={footerClickEventAction}
								eventLabel="Freedom of information"
							>
								Freedom of information
							</TrackedLink>
						</li>
						<li>
							<TrackedLink
								href="https://www.nice.org.uk/glossary"
								eventAction={footerClickEventAction}
								eventLabel="Glossary"
							>
								Glossary
							</TrackedLink>
						</li>
						<li>
							<TrackedLink
								href="https://www.nice.org.uk/terms-and-conditions"
								eventAction={footerClickEventAction}
								eventLabel="Terms and conditions"
							>
								Terms and conditions
							</TrackedLink>
						</li>
						<li>
							<TrackedLink
								href="https://www.nice.org.uk/privacy-notice"
								eventAction={footerClickEventAction}
								eventLabel="Privacy notice"
							>
								Privacy notice
							</TrackedLink>
						</li>
						<li>
							<TrackedLink
								href="https://www.nice.org.uk/cookies"
								eventAction={footerClickEventAction}
								eventLabel="Cookies"
							>
								Cookies
							</TrackedLink>
						</li>
					</ul>
				</nav>
				<p className={styles.copyright}>
					&copy;{" "}
					<abbr title="National Institute for Health and Care Excellence">
						NICE
					</abbr>{" "}
					{new Date().getFullYear()}. All rights reserved. Subject to{" "}
					<TrackedLink
						href="https://www.nice.org.uk/terms-and-conditions#notice-of-rights"
						eventAction={footerClickEventAction}
						eventLabel="Notice of rights"
					>
						Notice of rights
					</TrackedLink>
					.
				</p>
			</div>
		</div>
	);
};

export default Legal;
