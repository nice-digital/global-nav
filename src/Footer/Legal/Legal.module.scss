@use '../../styles/global-nav-colours' as *;
@use '@nice-digital/nds-core/scss/layout';
@use '@nice-digital/nds-core/scss/media-queries';
@use '@nice-digital/nds-core/scss/spacing';
@use '@nice-digital/nds-core/scss/typography';
@use '@nice-digital/nds-core/scss/utils';
@use '@nice-digital/nds-core/scss/visibility';

.wrapper {
  @include typography.font-size(-1);
  background: $legal-background;
  padding: utils.rem(spacing.$medium 0);

  a {
    text-decoration: underline;
  }
}

.container {
  @include layout.container;
}

.menu {
  @include visibility.hide-print;
  margin: utils.rem(0 0 spacing.$medium);

  ul {
    list-style: none;
    margin: 0;
    max-width: 100%;
    padding: 0;
  }

  li {
    margin: 0;
  }

  a {
    display: block;
  }
}

.copyright {
  @include typography.font-size(-1);
  margin: 0;
  @media print {
    float: none !important;
    max-width: 100%;
  }
}

@include media-queries.mq($from: sm) {
  .menu {
    li {
      display: inline-block;
      margin-right: utils.rem(spacing.$medium);
    }
  }
}

@include media-queries.mq($from: lg) {
  .wrapper {
    @include layout.clearfix;
  }

  .menu {
    float: left;
    margin: 0;
  }

  .copyright {
    float: right;
  }
}
