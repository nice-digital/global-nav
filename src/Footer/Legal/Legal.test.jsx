import MockDate from "mockdate";

import Legal from "./Legal";
import { render } from "@testing-library/react";

describe("Legal", () => {
	it("Matches snapshot", () => {
		MockDate.set("2000-11-22");
		const { container } = render(<Legal />);
		expect(container).toMatchSnapshot();
		MockDate.reset();
	});
});
