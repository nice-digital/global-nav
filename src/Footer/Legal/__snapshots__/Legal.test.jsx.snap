// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legal Matches snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="container"
    >
      <nav
        aria-label="Legal menu"
        class="menu"
      >
        <ul>
          <li>
            <a
              href="https://www.nice.org.uk/accessibility"
            >
              Accessibility
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/freedom-of-information"
            >
              Freedom of information
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/glossary"
            >
              Glossary
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/terms-and-conditions"
            >
              Terms and conditions
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/privacy-notice"
            >
              Privacy notice
            </a>
          </li>
          <li>
            <a
              href="https://www.nice.org.uk/cookies"
            >
              Cookies
            </a>
          </li>
        </ul>
      </nav>
      <p
        class="copyright"
      >
        Â©
         
        <abbr
          title="National Institute for Health and Care Excellence"
        >
          NICE
        </abbr>
         
        2000
        . All rights reserved. Subject to
         
        <a
          href="https://www.nice.org.uk/terms-and-conditions#notice-of-rights"
        >
          Notice of rights
        </a>
        .
      </p>
    </div>
  </div>
</div>
`;
